### DISCUSSION TITLE
Distribution of donation: `2023/06/01 ~ 2024/12/3`

### DISCUSSION THREAD

**Discussion by kdy1** [swc-project/swc#9779] [General]
- **NOTE**: **I modified the raw contribution score data** because the original version did not account for the PR content. The new script now uses the list of changed files of each PRs while asking GPT for the score of the PR.

 - **NOTE**: **I modified the raw contribution score data** again because the previous script does not account for the diff size. I added the number of changed lines along with the path to the file.

---

I wrote a script and collected all data from github in the given period. After than, I processed them through the openai API. (gpt-4o-mini) to give each activity different point. 

Current balance is `$13,825.65`, and if I remember correctly there was a fee of 10%. So **I'll distribute `$12000`** to be safe. 

My plan is to 

 - exclude me
 - give `3x` points to core team members, because they helped me maintaining a lot. I think this can be justified by excluding me, considering the contribution score, and the fact that I can ask them (before they says to) to resolve issue really helps maintenance.
 - give 1x point to external contributors
 - give the same ratio of $12000 as the ratio of the point : total point
 - if someone do not want it, their points will be excluded from the total point

**I'll distribute this starting from 2024/12/9**.

 - **If you don't agree with anything here or have a question, please leave a comment.**

My english is not so good, so feel free to ask for elaboration if you need.

I'll attach the raw contribution score.

Core team:
 - @kdy1 
 - @magic-akari 
 - @Austaras
 - @Brooooooklyn
 - @CPunisher
 - @kwonoj

<details>
<summary>Raw contribution score</summary>

User Contribution Scores:
=======================
User: @kdy1
Total Score: 15375.00
------------------------
User: @magic-akari
Total Score: 820.00
------------------------
User: @Brooooooklyn
Total Score: 587.00
------------------------
User: @dsherret
Total Score: 411.00
------------------------
User: @Austaras
Total Score: 330.00
------------------------
User: @zhangtao25
Total Score: 274.00
------------------------
User: @hyp3rflow
Total Score: 250.00
------------------------
User: @devjiwonchoi
Total Score: 248.00
------------------------
User: @fargito
Total Score: 210.00
------------------------
User: @yeliex
Total Score: 169.00
------------------------
User: @CPunisher
Total Score: 166.00
------------------------
User: @levi-nz
Total Score: 156.00
------------------------
User: @ciffelia
Total Score: 156.00
------------------------
User: @noyobo
Total Score: 149.00
------------------------
User: @hardfist
Total Score: 149.00
------------------------
User: @andersk
Total Score: 137.00
------------------------
User: @davidmurdoch
Total Score: 131.00
------------------------
</details>

For old data without PR content analysis, please see https://gist.github.com/kdy1/2f7be89d3fc16e18c447cffded7fc389

**Comment by ted-marozzi** [comment]
Why exclude yourself?

**Reply by kdy1** [reply]
It will make my tax issues even more complex. Also, my tax rate is very high so it does not help.

**Reply by ted-marozzi** [reply]
Donation's are taxed? 😢 Fair enough.

**Comment by travzhang** [comment]
How to pay?

**Reply by kdy1** [reply]
You can request it via open collective. https://opencollective.com/swc/expenses/new

**Comment by kdy1** [comment]
I think I may need to adjust the contribution calculator script to account for PR size. I didn't read the raw result carefully, but it feels strange, and the cause is a lack of PR size analysis.

Currently, PR is weighted over issues or comments, but every PR has same range of min/max scores.

**Comment by ArturAralin** [comment]
@kdy1 I'm really appreciate, but you can exclude me from the list?

**Reply by kdy1** [reply]
I did it :)

**Comment by simonbuchan** [comment]
Exclude me, too, please!

**Reply by kdy1** [reply]
Done 👍

**Comment by desout** [comment]
Thank you very much.
Is there a brief explanation of how this works? I seem to be new here and don't know what the next step is?
Or poke me in the right documentation

**Reply by kdy1** [reply]
You can just wait until 12/9, and I'll request each user to send a pay request via https://opencollective.com/swc/expenses/new

**Comment by silverwind** [comment]
Exclude me as well.

**Reply by kdy1** [reply]
Done

**Comment by frankpengau** [comment]
Thank you for the recognition but I don't believe my contributions are worth splitting OSS donations. 

You can exclude me from the calculations! :)

**Reply by kdy1** [reply]
Done 👍

**Comment by 9aoy** [comment]
Exclude me too, thanks~

**Reply by kdy1** [reply]
Done 👍

**Comment by sgomes** [comment]
Hello! 👋

Please exclude me as well, as my only contribution was a simple one-liner documentation fix 🙂 Any OSS donations are better spent on other community members. Thank you!

**Reply by kdy1** [reply]
Done 👍

**Comment by Codex-** [comment]
This is incredibly generous! I've registered on Open Collective now: https://opencollective.com/codex-nz

So we wait until 9th December and you'll send the details of what needs to be requested?

**Reply by kdy1** [reply]
Yeap, exactly.

**Reply by desout** [reply]
I just checked.
Me neither.
Potentially, the right calculation works now
It doesn't matter.

**Comment by leafypout** [comment]
So is the amount of points the amount of money you will receive?

**Reply by kdy1** [reply]
Nope, `$12000 * ${your point} / ${total point}` is the amount of the money you will receive.

**Comment by MasterOdin** [comment]
Please exclude me as my contribution was super minimal/simple.

**Reply by kdy1** [reply]
Done

**Comment by abnersajr** [comment]
Please remove my name.

Well. I have no idea on how my name is in this list. But anyway, as this is a free and open-source project you can exclude my name and distribute my points across the core team members.

**Reply by kdy1** [reply]
Done

**Comment by kdy1** [comment]
I think the problem is that I didn't add the numbers of changed lines while throwing them to GPT for the score.

**Comment by paolobarbolini** [comment]
My contribution was too a super minimal thing. You can exclude it.

**Reply by kdy1** [reply]
Done

**Comment by szubster** [comment]
I do not think I have contributed to this project.
https://github.com/swc-project/swc/commits/main/?author=szubster

**Reply by kdy1** [reply]
Yeah, the PR is https://github.com/swc-project/jest/pull/170, but I think the script should account for the number of lines that have been changed.

**Reply by szubster** [reply]
Please exclude me and redistribute to others :)

**Comment by vitorhugomattos** [comment]
I just fixed a typo in the docs, please exclude me from the list too.

**Reply by kdy1** [reply]
Done. Thank you!

**Comment by vincentlepot** [comment]
Hi, thanks for the attention, but I can't even remember contributing to the project. You can remove my name and redistribute my points to others.

**Reply by kdy1** [reply]
Done. Thank you!

**Comment by JasonFyfe** [comment]
You can redistribute my points too to other contributors. Thank you!

**Reply by kdy1** [reply]
Done, thanks!

**Comment by mertcanaltin** [comment]
you can keep me out, my contribution was very little

**Reply by mertcanaltin** [reply]
https://github.com/swc-project/swc/pull/9441

**Reply by kdy1** [reply]
Thanks!

**Comment by marco-ippolito** [comment]
Exclude me too 💚

**Reply by kdy1** [reply]
Done 👍

**Comment by ShogunPanda** [comment]
@kdy1 I love what you're trying to do but my contribution was close to nothing. Definitely exclude me from this!

**Reply by ShogunPanda** [reply]
Nevermind, I think I was just included in the first draft. Nothing to do here XD

**Comment by SukkaW** [comment]
You can keep me out as well. I have only submitted a few PRs toward `@swc/register` alongside a few type declaration improvements.

**Reply by kdy1** [reply]
Done 👍

**Comment by meskill** [comment]
I was tagged at some point but don't see my name in the current list. Was it removed because it's insignificant?

**Reply by kdy1** [reply]
I guess so.

**Comment by JSerFeng** [comment]
You can exclude me as well

**Reply by kdy1** [reply]
Done

**Comment by personalcodecare** [comment]
Can I participate in this project?

**Reply by kdy1** [reply]
Of course

**Comment by kdy1** [comment]
Dear contributors, you can see the amount you can request at https://swc-donation-1.vercel.app/

After checking it, please go to https://opencollective.com/swc/expenses/new and submit a request if you are in the list.

**Reply by Codex-** [reply]
I'm a bit confused, I was on the list but am no longer on the list, is this because of the re-calculations?

**Reply by kdy1** [reply]
Yeah, I think so. The new script account for the size of changes in each files to ensure contributors who made large PRs are rewarded fairly.

**Reply by Codex-** [reply]
Understandable, thanks for the clarification ✨

**Reply by kdy1** [reply]
Thank you for understanding!

**Reply by yeliex** [reply]
should I calc the final amount I can award before submitting?

**Reply by kdy1** [reply]
@yeliex See https://swc-donation-1.vercel.app/

**Reply by komyg** [reply]
Hey @kdy1, I am getting a 404 error on the donation app. Is it still online?

**Reply by kdy1** [reply]
It seems like I removed it accidentally while cleaning vercel projects up.

**Comment by kdy1** [comment]
I approved all verifiable requests. 

Left:

GitHub | score | point | amount
-- | -- | -- | --
@dsherret | 411.00 | 411.00 | $605.23

Please use an open collective account that's connected to your GitHub username. If you can't please left a comment on this thread.

**Reply by travzhang** [reply]
This is my open collective account: zhangtao251

**Reply by kdy1** [reply]
Approved 👍.
Thank you all, contributors.

**Reply by leafypout** [reply]
My username is: levi-michael-taylor

Also please make sure I requested the right amount

**Reply by kdy1** [reply]
@levi-nz Your reward should be $229.72. Currently, it's $156.00. I tried editing but I couldn't because I don't know the place.

**Reply by leafypout** [reply]
> @levi-nz Your reward should be $229.72. Currently, it's $156.00. I tried editing but I couldn't because I don't know the place.

Fixed

**Reply by fargito** [reply]
Hello, my open collective account is: fargito. Thanks a lot!

**Reply by kdy1** [reply]
Done 👍

**Reply by kdy1** [reply]
@hyp3rflow Are you using different account for open collective?

**Reply by hyp3rflow** [reply]
@kdy1 yes, my account is leo-choi.

**Reply by kdy1** [reply]
Done

**Reply by kdy1** [reply]
@Austaras You can request a lot more. You requested `$187.55 USD` but you can request $1457.85

**Reply by Austaras** [reply]
Oh I use the wrong currency.

**Reply by hardfist** [reply]
thanks, i request via rspack account

**Reply by Austaras** [reply]
I've updated my request.

**Reply by kdy1** [reply]
Approved 👍

**Reply by dsherret** [reply]
Thanks!! I submitted via username david-sherret and also opened another PR (though I was going to do that anyway) :D

**Reply by kdy1** [reply]
Approved, thank you all for the contributions!
